packages/hdwallet-native/src/crypto/engines/web-crypto.ts

```diff
@@ -1,15 +1,20 @@
-import { CryptoEngine, ScryptParams } from "./index";
 import { scrypt } from "scrypt-js";
+import { CryptoEngine, DigestAlgorithm, ScryptParams } from "./index";
 
-export default class WebCryptoEngine implements CryptoEngine {
+export class WebCryptoEngine implements CryptoEngine {
   public async decrypt(data: ArrayBuffer, key: ArrayBuffer, iv: ArrayBuffer): Promise<ArrayBuffer> {
     const impKey = await globalThis.crypto.subtle.importKey("raw", key, { name: "AES-CBC" }, false, ["decrypt"]);
-    return await globalThis.crypto.subtle.decrypt({ name: "AES-CBC", iv }, impKey, data);
+    return globalThis.crypto.subtle.decrypt({ name: "AES-CBC", iv }, impKey, data);
+  }
+
+  public async digest(algorithm: DigestAlgorithm, data: ArrayBuffer): Promise<ArrayBuffer> {
```

Comment:
this need to be async?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest

> digest is a Promise that fulfills with an ArrayBuffer containing the digest.

Reply:
Also, there is a parent contract defined called CryptoEngine in which all the functions are declared `async` because it's easier to return a synchronous result as a Promise than the other way around. Currently, both the web crypto and React Native crypto use async functions for all crypto code.